JS/marquee.js

```diff
@@ -0,0 +1,40 @@
+class Marquee {
+    constructor(element) {
+        this.element = element;
+        this.url = `https://stock-exchange-dot-full-stack-course-services.ew.r.appspot.com/api/v3/quotes/nyse/`
+    }
+
+    async load() {
+        this.data = await this.getData();
+        this.presentData();
+      }
+   
+    async getData() {
+        try {
+            const response = await fetch(this.url);
+            const result = await response.json();
+            return result;
+        }
+        catch (err) {
+            throw new Error("ERROR: server error", err);
+        }
+    }
+
+    async presentData() {
+        let marqueeElement = this.element;
+        marqueeElement.className = "marqueeElement flex-row";
+        let marqueeItems = document.createElement("div")
+        marqueeItems.className = "marquee";
+        marqueeElement.appendChild(marqueeItems);
+
+        let slicedResult = this.data.slice(0,200);
+            slicedResult.forEach(item => {
+                let symbol = item.symbol;
+                let price = item.open;
+                marqueeItems.innerHTML+=`<span class="marquee-item"> <b>${symbol}</b> <span class="marquee-price">${price}</span></span>` 
```

Comment:
Although you code works, DOM manipulation is an action that requires a lot of computing power (DOM access is "expensive" for the browser) and since then it should be always done outside of loop iteration. A better approach would be to aggregate all the new elements inside a parent element that does not live inside the DOM yet, and only after the loop iteration is done you should append the parent element with all the children inside of it. You can also check about createDocumentFragment() (google it!)

Reply:
Got it, implemented in Milestone-6